### PR TITLE
Defines a new CRD for the preview features (attempt 2)

### PR DESCRIPTION
**Pull request by m1kola**
Note: I had to revert #1670 from master because it was relying on operator SDK.

### Which issue this PR addresses:

Part of work item [№10340376](https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/10340376/).

### What this PR does / why we need it:

Adds new types for preview feature controller (includes NSG flow log configuration).

### Test plan for issue:

Unit tests/e2e should catch regressions. New types are not being used yet.

### Is there any documentation that needs to be updated for this PR?

No new features introduced yet.

### CODE REVIEW

File: pkg/operator/apis/preview.aro.openshift.io/v1alpha1/previewfeature_types.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	// NSGFlowLogs contains configuration for NSG flow logs.
+	// Omit the configuration if you don't want the controller
+	// to reconcile NSG flow logs.
+	NSGFlowLogs *NSGFlowLogs `json:"nsgFlowLogs,omitempty"`
+}
+
+// PreviewFeatureStatus defines the observed state of PreviewFeature
+type PreviewFeatureStatus struct {
+	OperatorVersion   string                         `json:"operatorVersion,omitempty"`
+	Conditions        []operatorv1.OperatorCondition `json:"conditions,omitempty"`
```
Comment (m1kola) [inline pkg/operator/apis/preview.aro.openshift.io/v1alpha1/previewfeature_types.go:56]: This line and import in this file were the only chance I made manually in comparison to orignal PR #1670.

This channge caused auto-generated code to be a bit different, but that was expected.

**Review by jewzaam — APPROVED** [review]
Only thought is on the apigroup being "preview.aro.openshift.io", the resource name containing "preview" and ARO having "preview" RP API's.. just a bit confusing.  Is there a value in having this in a new apigroup or could it be in "aro.openshift.io"?

**Comment by m1kola** [comment]
> Only thought is on the apigroup being "preview.aro.openshift.io", the resource name containing "preview" and ARO having "preview" RP API's.. just a bit confusing. Is there a value in having this in a new apigroup or could it be in "aro.openshift.io"?

@jewzaam the idea is that we split SRE CRDs from ones which are going to be managed by customers into separate groups:

* `aro.openshift.io` - existing one for SREs
* `preview.aro.openshift.io` - for customers.

**Pull request merged**